src/lib/y2security/clients/security_policy_proposal.rb

```diff
@@ -323,6 +247,164 @@ def storage_client
       ensure
         Yast::Wizard.CloseDialog
       end
+
+      # Builds links for a rule in the context of a dialog
+      class LinksBuilder
+        def initialize(dialog_id)
+          @dialog_id = dialog_id
+        end
+
+        def links_for_policy(policy)
+          [policy_toggle_link(policy)]
+        end
+
+        def links_for_rule(rule)
+          [
+            rule_toggle_link(rule),
+            rule_fix_link(rule)
+          ]
+        end
+
+        def policy_toggle_link(policy)
+          build_link("toggle-policy", policy.id)
+        end
+
+        def rule_toggle_link(rule)
+          build_link("toggle-rule", rule.id)
+        end
+
+        def rule_fix_link(rule)
+          if rule.fixable?
+            build_link("fix-rule", rule.id)
+          elsif rule.scope == :storage
+            build_link("storage")
+          end
+        end
+
+      private
+
+        attr_reader :dialog_id
+
+        # Builds a link
+        #
+        # @param action [String] Action ("enable", "disable" or "fix")
+        # @param id [#to_s] id of the element to act on
+        # @return [String]
+        def build_link(action, id = nil)
+          "#{dialog_id}--#{action}:#{id}"
+        end
+      end
+
+      # Builds the representation of a security policy
+      class PolicyPresenter
+        include Yast::I18n
+
+        def initialize(policy, enabled:, failing_rules:, links_builder:)
```

Comment:
```suggestion
        # @param policy [SecurityPolicies::Policy]
        # @param failing_rules [Array<SecurityPolicies::Rule>]
        # @param links_builder [LinksBuilder]
        def initialize(policy, enabled:, failing_rules:, links_builder:)
```

failing_rules is a bit confusing because elsewhere it is a hash of arrays